tensorflow/python/lib/io/file_io.py

```diff
@@ -587,6 +590,41 @@ def is_directory_v2(path):
     return False
 
 
+@tf_export(v1=["gfile.NeedsTempLocation"])
+def needs_temp_location(path):
+  """ Returns whether or not writing to the given path needs to use
+      a temporary location for safety
+  Args:
+    path: string, path to a file
+  Returns:
+    True, if the path is on a file system that needs to use a temporary
+          location to write safely. In such cases it is recommended to write to
+          a temporary location and then do (atomic) move to the final location.
+    False, if it is safe to write to the path without a temp location
+  """
+  return needs_temp_location_v2(path)
+
+
+@tf_export("io.gfile.needstemp")
+def needs_temp_location_v2(path):
+  """ Returns whether or not writing to the given path needs to use
+      a temporary location for safety
```

Comment:
It's not clear what this API would be used for; can you provide an example of how a user would use this? ( https://www.tensorflow.org/community/contribute/docs_ref#testable_docstrings ) 

Further, we are a little confused by the name here-- it seems that this is really checking whether renames are atomic; can we name and document according to the use-case rather than the solution?

Reply:
Would has_atomic_move be a better name for this function? It is used by other parts of the TF codebase when checkpointing/saving models, to determine whether a temporary location should be used or not.

Reply:
Is there a way to expose this function to the python parts of TF without necessarily making this a top level API for users?

Reply:
If you just leave off the tf_export decorator this is still usable inside TensorFlow, it just won't be in the tf.* API. We don't use tf.* in TF's Python implementation or unit tests anyway.

Reply:
Or should the name be is_strongly_consistent, as in Is this a strongly consistent filesystem? That's basically what's the consideration here rather than has_atomic_move

Reply:
Thanks, done!